code/datums/components/attachment_handler.dm

```diff
@@ -354,7 +354,9 @@
 		var/icon = attachment.icon
 		var/icon_state = attachment.icon_state
 		var/suffix = ""
-		if(!CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_SAME_ICON))
+		if(CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_DIFFERENT_MOB_ICON_STATE))
+			suffix = "_m"
+		else if(!CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_SAME_ICON))
```

Comment:
?

Reply:
This is for the beret badges since they are now colorable. Did i mention that in the pr? I'll do that.
Since the badge has two different versions, one for inhand and one for on the mob it needs to change states when attached, however since it is greyscale it needs to use the same icon. We didnt have logic for it so thats what this is.

Reply:
atomize next time